docs/guides/exporters/index.md

```diff
@@ -1,103 +1,218 @@
+[`--output` CLI option](../../reference/buildx_build.md#output). Buildx supports
+the following exporters:
+
+- `image`: exports the build result to a container image.
+- `registry`: exports the build result into a container image, and pushes it to
+  the specified registry.
+- `local`: exports the build root filesystem into a local directory.
+- `tar`: packs the build root filesystem into a local tarball.
+- `oci`: exports the build result to the local filesystem in the
+  [OCI image layout](https://github.com/opencontainers/image-spec/blob/v1.0.1/image-layout.md)
+  format.
+- `docker`: exports the build result to the local filesystem in the
+  [Docker image](https://github.com/docker/docker/blob/v20.10.2/image/spec/v1.2.md)
+  format.
+- `cacheonly`: doesn't export a build output, but runs the build and creates a
+  cache.
 
-...
+## Using exporters
 
-Buildx supports the following exporters:
+To specify an exporter, use the following command syntax:
 
-- `image` / `registry`: exports the build result into a container image.
-- `local`: exports the build root filesystem into a local directory.
-- `tar`: packs the build root filesystem into a local tarball.
-- `oci`: exports the build result as a local
-  [OCI image layout](https://github.com/opencontainers/image-spec/blob/v1.0.1/image-layout.md).
-- `docker`: exports the build result as a local
-  [Docker image specification](https://github.com/docker/docker/blob/v20.10.2/image/spec/v1.2.md).
+```console
+$ docker buildx build --tag <registry>/<image> \
+  --output type=<TYPE> .
+```
+
+Most common use cases doesn't require you don't need to specify which exporter
+to use explicitly. You only need to specify the exporter if you intend to
+customize the output somehow, or if you want to save it to disk. The `--load`
+and `--push` options allow Buildx to infer the exporter settings to use.
+
+For example, if you use the `--push` option in combination with `--tag`, Buildx
+automatically uses the `image` exporter, and configures the exporter to push the
+results to the specified registry.
+
+To get the full flexibility out of the various exporters BuildKit has to offer,
+you use the `--output` flag that lets you configure exporter options.
+
+## Use cases
+
+Each exporter type is designed for different use cases. The following sections
+describe some common scenarios, and how you can use exporters to generate the
+output that you need.
+
+### Load to image store
+
+Buildx is often used to build container images that can be loaded to an image
+store. That's where the `oci` and `docker` exporters come in. The following
+example shows how to build an image using the `oci` exporter, and have that
+image loaded to the local image store, using the `--output` option:
+
+```console
+$ docker buildx build \
+  --output type=oci,name=<registry>/<image>,store=true .
+```
+
+Buildx CLI will automatically use the `oci` exporter and load it to the image
+store if you provide the `--tag` and `--load` options:
```

Comment:
I think we want to talk about `type=docker` here, instead of `type=oci,store=true`.